### PR TITLE
(FACT-2859) Implement load_external API method

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Comment by oanatmaria** [comment]
Jenkins please test this on all

**Review by florindragos — APPROVED** [review]

File: acceptance/tests/facter_load_external.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+test_name "Facter should not diaplay external facts when 'load_external' method is called with false"  do
```
Comment (Filipovici-Andrei) [inline acceptance/tests/facter_load_external.rb:1]: typo in display

File: acceptance/tests/facter_load_external.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    step "Agent #{agent}: create facts.d directory and fact" do
+      agent.mkdir_p(factsd)
+      create_remote_file(agent, ext_fact, content)
+      agent.chmod('+x', ext_fact)
+    end
+
+    step "Agent #{agent}: ensure that external fact is loaded and resolved" do
+      on(agent, "#{Puppet::Acceptance::CommandUtils.ruby_command(agent)} #{script_name2}", :acceptable_exit_codes => 1)
```
Comment (Filipovici-Andrei) [inline acceptance/tests/facter_load_external.rb:57]: Puppet::Acceptance::CommandUtils.ruby_command does the same thing as simple ruby_command from beaker-puppet. We should prefer the latter.

File: lib/facter/framework/core/options/option_store.rb [hunk 3]
```diff
@@ -178,6 +178,7 @@ def reset
         @block = true
         @cli = nil
         @custom_facts = true
+        @external_facts = true
```
Comment (Filipovici-Andrei) [inline lib/facter/framework/core/options/option_store.rb:181]: external_facts is being reset in the reset_config method

File: spec/facter/facter_spec.rb [hunk 4]
```diff
@@ -435,6 +435,25 @@ def mock_collection(method, os_name = nil, error = nil)
       end
     end
 
+    describe '#load_external' do
+      before do
```
Comment (Filipovici-Andrei) [inline spec/facter/facter_spec.rb:439]: probably we should add a test when the external facts are not loaded

**Review by Filipovici-Andrei — CHANGES_REQUESTED** [review]

File: spec/facter/facter_spec.rb (hunk 4 above)
Comment (oanatmaria) [reply spec/facter/facter_spec.rb:439]: There already is an acceptance test added with both scenarios.

**Pull request merged**